libraries/stdlib/jvm/src/kotlin/text/StringsJVM.kt

```diff
@@ -424,8 +424,14 @@ public inline fun String.toPattern(flags: Int = 0): java.util.regex.Pattern {
  *
  * @sample samples.text.Strings.capitalize
  */
-public actual fun String.capitalize(): String {
-    return if (isNotEmpty() && this[0].isLowerCase()) substring(0, 1).toUpperCase() + substring(1) else this
+public actual fun String.capitalize() = capitalize(Locale.getDefault())
+
+/**
+ * Returns a copy of this string having its first letter uppercased using [locale], or the original string,
+ * if it's empty or already starts with an upper case letter.
+ */
+public fun String.capitalize(locale: Locale): String {
```

Comment:
The added functions should be annotated with `@SinceKotlin("1.3")` and the new experimental annotation `@ExperimentalStdlibApi`, so we could publish them in 1.3.x release.